### ISSUE TITLE
Emit detailed error messages for 4xx responses

### ISSUE THREAD

**Issue by webignition** [smartassert/api-gateway#191]
The user-level consumers of the error responses of this application will need to provide detailed errors to uses. This can be supported by, or made possible by, the provision of detailed 4xx response bodies.

Will need to consider how to present machine-readable error detail such that it can act as source for both machine- and human-readable error content.

This will likely then be factored out to a package used by some (all?) services on which the api gateway depends.

**Comment by webignition** [comment]
Example for empty field:

```json
{
  "type": "invalid_request_field",
  "field": {
    "name": "label",
    "value": ""
  },
  "error": {
    "class": "empty"    
  }
}
```

**Comment by webignition** [comment]
Duplicate unique value:

```json
{
  "class": "invalid_request_field",
  "type": "duplicate",
  "field": {
    "name": "label",
    "value": "Value must be unique and is already taken"
  }
}
```

**Comment by webignition** [comment]
Invalid value data type (400 response):

```json
{
  "class": "invalid_request_field",
  "field": {
    "name": "count",
    "value": "string value",
    "data-type": "string"
  },
  "type": "incorrect-data-type",
  "requirements": {
    "data_type": "int"
  }
}
```

**Comment by webignition** [comment]
Fixed by #202
